src/textual/_segment_tools.py

```diff
@@ -58,3 +63,64 @@ def line_crop(segments: list[Segment], start: int, end: int, total: int):
         segment = next(iter_segments, None)
 
     return output_segments
+
+
+def line_trim(segments: list[Segment], start: bool, end: bool) -> list[Segment]:
+    """Optionally remove a cell from the start and / or end of a list of segments.
+
+    Args:
+        segments (list[Segment]): A line (list of Segments)
+        start (bool): Remove cell from start.
+        end (bool): Remove cell from end.
+
+    Returns:
+        list[Segment]: A new list of segments.
+    """
+    segments = segments.copy()
+    if segments and start:
+        _, first_segment = segments[0].split_cells(1)
+        if first_segment.text:
+            segments[0] = first_segment
+        else:
+            segments.pop(0)
+    if segments and end:
+        last_segment = segments[-1]
+        last_segment, _ = last_segment.split_cells(len(last_segment.text) - 1)
+        if last_segment.text:
+            segments[-1] = last_segment
+        else:
+            segments.pop()
+    return segments
+
+
+def line_pad(
+    segments: Iterable[Segment], pad_left: int, pad_right: int, style: Style
+) -> list[Segment]:
+    """Adds padding to the left and / or right of a list of segments.
+
+    Args:
+        segments (list[Segment]): A line (list of Segments).
+        pad_left (int): Cells to pad on the left.
+        pad_right (int): Cells to pad on the right.
+        style (Style): Style of padded cells.
+
+    Returns:
+        list[Segment]: A new line with padding.
+    """
+    if pad_left and pad_right:
+        segments = [
+            Segment(" " * pad_left, style),
+            *segments,
+            Segment(" " * pad_right, style),
+        ]
+    elif pad_left:
+        segments = [
+            Segment(" " * pad_left, style),
+            *segments,
+        ]
+    elif pad_right:
+        segments = [
+            *segments,
+            Segment(" " * pad_right, style),
+        ]
+    return list(segments)
```

Comment:
[trivial comment] I guess using _early returns_ would avoid having to create a new list object at the end just to make MyPy happy? :slightly_smiling_face: 

```suggestion
    if pad_left and pad_right:
        return [
            Segment(" " * pad_left, style),
            *segments,
            Segment(" " * pad_right, style),
        ]
    if pad_left:
        return [
            Segment(" " * pad_left, style),
            *segments,
        ]

    return [
        *segments,
        Segment(" " * pad_right, style),
    ]

```

Reply:
Good point. Although it would still need one case if neither pad is set.